packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.js

```diff
@@ -42,48 +42,44 @@ class Search extends React.Component {
       });
     } else {
       console.warn('Search has failed to load and now is being disabled');
-      this.setState({enabled: false});
+      setIsEnabled(false);
     }
-  }
-
-  toggleSearchIconClick() {
-    this.setState(
-      oldState => ({
-        isExpanded: !oldState.isExpanded,
-      }),
-      () => {
-        this.searchBarRef.current.focus();
-        this.props.handleSearchBarToggle();
-      },
-    );
-  }
+  }, []);
 
-  render() {
-    const {enabled, isExpanded} = this.state;
+  const toggleSearchIconClick = useCallback(() => {
+    props.handleSearchBarToggle(!props.isSearchBarExpanded);
+  }, [props.isSearchBarExpanded]);
 
-    return enabled ? (
-      <Fragment>
-        <span
-          role="button"
-          className={`search-icon ${isExpanded ? 'search-icon-hidden' : ''}`}
-          onClick={this.toggleSearchIconClick}
-          onKeyDown={this.toggleSearchIconClick}
-          tabIndex={0}
-        />
-        <input
-          id="search_input_react"
-          type="search"
-          placeholder="Search"
-          aria-label="Search"
-          className={`${isExpanded ? 'search-bar-expanded' : 'search-bar'}`}
-          onBlur={this.toggleSearchIconClick}
-          ref={this.searchBarRef}
-        />
-      </Fragment>
-    ) : null;
-  }
-}
+  useEffect(() => {
+    if (props.isSearchBarExpanded) {
+      searchBarRef.current.focus();
+    }
+  }, [props.isSearchBarExpanded]);
 
-Search.contextType = DocusaurusContext;
+  return isEnabled ? (
+    <Fragment>
+      <span
+        role="button"
+        className={`search-icon ${
```

Comment:
Use `classnames` package for conditional CSS rendering please. Refer to the other examples in the project.